benchmark_runner/common/oc/oc.py

```diff
@@ -82,26 +75,29 @@ def get_worker_disk_ids(self):
         """
         workers_disk_ids = []
         if self.__worker_disk_ids:
-            workers_disk_ids = [disk_id for disk_list in self.__worker_disk_ids.values() for disk_id in disk_list]
+            for node, disk_ids in self.__worker_disk_ids.items():
+                for disk_id in disk_ids:
+                    workers_disk_ids.append(disk_id)
         return workers_disk_ids
 
+    def get_pv_disk_ids(self):
+        """
+        This method returns list of pv disk ids
+        """
+        pv_ids = self.run(f"{self.__cli} get pv -o jsonpath={{.items[*].metadata.annotations.'storage\.openshift\.com/device-id'}}")
+        return [pv[len(self.__worker_disk_prefix):] for pv in pv_ids.split()]
+
     def get_free_disk_id(self):
         """
-        This method finds the free disk by searching for worker_disk_ids that do not exist in the odf_pv_disk_ids list.
-        :return: It returns the free disk ID concatenated with the worker_disk_prefix
-        """
-        free_disk_id = ''
-        for worker_disk_id in self.get_worker_disk_ids():
-            found = False
-            for odf_pv_disk_id in self.get_pv_disk_ids():
-                if odf_pv_disk_id.endswith(worker_disk_id):
-                    found = True
-                    break
-            if not found:
-                free_disk_id = f'{self.__worker_disk_prefix}{worker_disk_id}'
-                break  # Exit the loop once a free disk ID is found
-
-        return free_disk_id
+        This method returns free disk (workers_all_disk_ids - workers_odf_pv_disk_ids) with its prefix
+        """
+        workers_disk_ids = self.get_worker_disk_ids()
+        workers_pv_disk_ids = self.get_pv_disk_ids()
+        free_disk_id = f'{self.__worker_disk_prefix}{list(set(workers_disk_ids) - set(workers_pv_disk_ids))[0]}'
+        if free_disk_id:
```

Comment:
The current code will always return something, because the prefix needs to be prepended only if you find a disk.

```
free_disk_id = {list(set(workers_disk_ids) - set(workers_pv_disk_ids))[0]}'
if free_disk_id:
    return f'{self.__worker_disk_prefix}{free_disk_id}'
else:
    raise Exception("No free disk found")
```

Reply:
OK